pkg/controller/master/virtualmachine/register.go

```diff
@@ -66,14 +67,17 @@ func Register(ctx context.Context, management *config.Management, options config
 	var virtualMachineCache = virtualMachineClient.Cache()
 	var virtualMachineInstanceClient = management.VirtFactory.Kubevirt().V1().VirtualMachineInstance()
 	var vmiCtrl = &VMIController{
+		vmClient:            management.VirtFactory.Kubevirt().V1().VirtualMachine(),
 		virtualMachineCache: virtualMachineCache,
 		vmiClient:           virtualMachineInstanceClient,
 		nodeCache:           nodeCache,
 		pvcClient:           pvcClient,
 		pvcCache:            pvcCache,
+		recorder:            management.NewRecorder(vmControllerSetHaltIfInsufficientResourceQuotaControllerName, "", ""),
 	}
 	virtualMachineInstanceClient.OnRemove(ctx, vmiControllerUnsetOwnerOfPVCsControllerName, vmiCtrl.UnsetOwnerOfPVCs)
 	virtualMachineInstanceClient.OnChange(ctx, vmiControllerReconcileFromHostLabelsControllerName, vmiCtrl.ReconcileFromHostLabels)
+	virtualMachineInstanceClient.OnChange(ctx, vmiControllerReconcileFromHostLabelsControllerName, vmiCtrl.SetHaltIfOccurExceededQuota)
```

Comment:
The controller name should be unique for each onChange method, don't use `vmiControllerReconcileFromHostLabelsControllerName`